src/Samples/SubModelSeq/Program.fs

```diff
@@ -13,177 +31,199 @@ module List =
         | a when a = j -> i
         | a -> a)
 
+  let swapWithNext i = swap i (i + 1)
+  let swapWithPrev i = swap i (i - 1)
+
+  let cons a ma = a :: ma
 
+        
 [<AutoOpen>]
-module Domain =
+module IdentifiableTypes =
+
+  type Identifiable<'a> =
+    { Id: Guid
+      Value: 'a }
+  
+  module Identifiable =
+    module Id =
+      let get m = m.Id
+      let set v m = { m with Id = v }
+      let map x = x |> map get set
+    module Value =
+      let get m = m.Value
+      let set v m = { m with Value = v }
+      let map f = f |> map get set
+      let mapIf p f = f |> map |> mapIf p
+      let mapIfId id f = f |> mapIf (fun m -> m.Id = id)
+  
+    let create v id =
+      { Id = id
+        Value = v }
 
-  type CounterId = CounterId of Guid
+
+[<AutoOpen>]
+module CounterTypes =
 
   type Counter =
-    { Id: CounterId
-      CounterValue: int
+    { Count: int
       StepSize: int }
 
+  type CounterMsg =
+    | Increment
+    | Decrement
+    | SetStepSize of int
+    | Reset
+
   module Counter =
 
-    let create () =
-      { Id = Guid.NewGuid () |> CounterId
-        CounterValue = 0
+    let init =
+      { Count = 0
         StepSize = 1 }
+    
+    let canReset = (<>) init
+    
+    let update msg m =
+      match msg with
+      | Increment -> { m with Count = m.Count + m.StepSize }
+      | Decrement -> { m with Count = m.Count - m.StepSize }
+      | SetStepSize x -> { m with StepSize = x }
+      | Reset -> init
 
-    let increment c =
-      { c with CounterValue = c.CounterValue + c.StepSize }
-
-    let decrement c =
-      { c with CounterValue = c.CounterValue - c.StepSize}
-
-    let setStepSize step c =
-      { c with StepSize = step }
 
-    let canReset c =
-      c.CounterValue <> 0 || c.StepSize <> 1
 
-    let reset c =
-      { create () with Id = c.Id }
+[<AutoOpen>]
+module TreeTypes =
 
+  type RoseTree<'a> =
+    { Data: 'a
+      Children: RoseTree<'a> list }
 
-module Tree =
 
-  type Node<'a> =
-    { Data: 'a
-      Children: Node<'a> list }
+  module RoseTree =
 
-  let asLeaf a =
-    { Data = a
-      Children = [] }
+    module Data =
+      let get m = m.Data
+      let set v m = { Data = v; Children = m.Children }
+      let rec map f n =
+        { Data = n.Data |> f
+          Children = n.Children |> (f |> map |> List.map) }
+    module Children =
+      let get m = m.Children
+      let set v m = { m with Children = v }
+      let map f = map get set f
```

Comment:
I suggest a similar change here as with `identifiable.Id/Value`. Rename and remove any unused functions.

Reply:
Sure.  I think all functions are used.